### ISSUE TITLE
React Native: Async Storage has been extracted from react native core. 

### ISSUE THREAD

**Issue by ponikar** [firebase/firebase-js-sdk#5786]
`firebase": ^9.5.0`

### Describe the problem
I am getting this warning whenever I am trying to use this method `onAuthStateChange` of  `firebase/auth`.

AsyncStorage has been extracted from `react-native` core and will be removed in a future release. It can now be installed and imported from `@react-native-async-storage/async-storage` instead of `react-native`. See https://github.com/react-native-async-storage/async-storage 

React Native no longer has AsyncStorage in core framework. Firebase needs to import it from `@react-native-async-storage/async-storage`

**Comment by jbalidiong** [comment]
> So the way it has to work because of the reasons discussed above is:
> 
> Firebase imports the old module from React-Native itself, triggering the warning
> The user sets their own persistence (@react-native-async-storage/async-storage)
> Firebase switches it's references to AsyncStorage to the user provided module
> This means A. that when AsyncStorage is removed firebase doesn't just up and stop working and B. firebase no longer conflicts with use of @react-native-async-storage/async-storage
> HOWEVER, firebase still has to import the old module to start with to support users who have not updated
> 
> TLDR; It does do something and you should use the new AsyncStorage even though it doesn't fix the warning

From https://github.com/firebase/firebase-js-sdk/issues/1847

See the link for more info and possible future plans.
